pkg/ingress2httpproxy/ingress2httpproxy_test.go

```diff
@@ -0,0 +1,52 @@
+package ingress2httpproxy
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	core "k8s.io/api/networking/v1beta1"
+)
+
+const clientName = "testClient"
+
+func TestBuildHP(t *testing.T) {
```

Comment:
Test function name should be `TestBuildHTTPProxy`. Also, would you be able to make these tests more Go-like? Here's a start:

```go
func TestBuildHTTPProxy(t *testing.T) {
	type output struct {
		httpProxy string
		ingress string
		apiVersion string
		secret string
		prefixes []string
		services []string
		ports []int
	}
	
	tests := []struct{
		name string
		input string
		want output
	}{
		{
			"Test config map with *.domain",
			"example.json",
			output{
				httpProxy: "HTTPProxy",
				ingress: "cafe-ingress",
				apiVersion: "v1beta1",
				secret: "cafe-secret",
				prefixes: []string{"/tea", "/coffee"},
				services: []string{"SVC1", "SVC2"},
				ports: []int{0, 0},
			},
		},
		{
			"Test config map with .domain",
			"example2.json",
			output{
				httpProxy: "HTTPProxy",
				ingress: "cafe-ingress",
				apiVersion: "v1beta1",
				secret: "cafe-secret",
				prefixes: []string{"/tea", "/coffee"},
				services: []string{"SVC1", "SVC2"},
				ports: []int{0, 0},
			},
		},
	}

	for _, tc := range tests {
		m := newMutatorFromFileData(t, tc.input)
		hp := m.buildHTTPProxy()

		assert.Equal(t, tc.want.httpProxy, hp.Kind)
		assert.Equal(t, tc.want.ingress, hp.Name)
		assert.Equal(t, tc.want.prefixes[0], hp.Spec.Routes[0].Conditions[0].Prefix)

		// and so forth
		
	}
}
```

I believe you had at least three examples you can try. 

Did you have any negative tests to try where you expect an error to happen? We can add those, too if you'd like. We'll need to add a field called "wantErr". If there is an error, we have a different if-statement to test the error to make sure it is the expected error. If it's not useful to test for errors, we can skip these tests.

Reply:
implemented the test logic as mentioned below.